docker-app/qfieldcloud/core/utils2/jobs.py

```diff
@@ -28,71 +29,78 @@ def apply_deltas(
     if not project.owner_can_create_job:
         return []
 
-    # 2. Check if there are any pending deltas.
-    # We need to call .select_for_update() to make sure there would not be a concurrent
-    # request that will try to apply these deltas.
-    pending_deltas = models.Delta.objects.select_for_update().filter(
-        project=project,
-        last_status=models.Delta.Status.PENDING,
-    )
-
-    # 2.1. Filter only the deltas of interest.
-    if len(delta_ids) > 0:
-        pending_deltas = pending_deltas.filter(pk__in=delta_ids)
-
-    # 3. If there are no pending deltas, do not create a new job and return.
-    if not pending_deltas.exists():
-        return []
-
-    # 4. Find all the pending or queued jobs in the queue.
-    # If an "apply_delta" job is in a "started" status, we don't know how far the execution reached
-    # so we better assume the deltas will reach a non-"pending" status.
-    apply_jobs = models.ApplyJob.objects.filter(
-        project=project,
-        status=[
-            models.Job.Status.PENDING,
-            models.Job.Status.QUEUED,
-        ],
-    )
-
-    # 5. Check whether there are jobs found in the queue and exclude all deltas that are part of any pending job.
-    if apply_jobs.exists():
-        pending_deltas = pending_deltas.exclude(jobs_to_apply__in=apply_jobs)
-
-    # 6. If there are no pending deltas, do not create a new job and return.
-    deltas_count = pending_deltas.count()
-    if deltas_count == 0:
-        return []
+    with advisory_lock(project.id):
+        # Ensure there is no concurrent request trying to apply these deltas.
```

Comment:
NOTE no code modified in the block body except removing the `select_for_update` .